azurerm_virtual_network_gateway crashes when removing vpn_client_configuration

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->



### Terraform Version

1.11.4

### AzureRM Provider Version

4.27.0

### Affected Resource(s)/Data Source(s)

azurerm_virtual_network_gateway

### Terraform Configuration Files

```hcl
resource "azurerm_virtual_network_gateway" "main" {
  name                                  = data.azurerm_resource_group.main.name
  location                              = data.azurerm_resource_group.main.location
  resource_group_name                   = data.azurerm_resource_group.main.name
  type                                  = "Vpn"
  vpn_type                              = "RouteBased"
  active_active                         = true
  enable_bgp                            = true
  sku                                   = var.vpn_sku
  bgp_route_translation_for_nat_enabled = (var.environment == "Production")
  dns_forwarding_enabled                = (var.environment == "Production" ? null : false)
  private_ip_address_enabled            = (var.environment == "Production" ? null : false)
  dynamic "vpn_client_configuration" {
    for_each = var.aad_audience[*]
    content {
      address_space        = [var.az_client_vpn_cidr]
      aad_audience         = var.aad_audience
      aad_issuer           = "https://sts.windows.net/${var.aad_tenant}/"
      aad_tenant           = "https://login.microsoftonline.com/${var.aad_tenant}"
      vpn_client_protocols = ["OpenVPN"]
      vpn_auth_types       = ["AAD"]
    }
  }
  # Cannot be changed after creation
  ip_configuration {
    name                 = "primary-vpn"
    public_ip_address_id = azurerm_public_ip.main["primary"].id
    subnet_id            = data.azurerm_subnet.vpn.id
  }
  ip_configuration {
    name                 = "secondary-vpn"
    public_ip_address_id = azurerm_public_ip.main["secondary"].id
    subnet_id            = data.azurerm_subnet.vpn.id
  }
  ip_configuration {
    name                 = "p2s-vpn"
    public_ip_address_id = azurerm_public_ip.main["p2s"].id
    subnet_id            = data.azurerm_subnet.vpn.id
  }
  bgp_settings {
    asn         = 65000
    peer_weight = (var.environment == "Production" ? null : 0)
    peering_addresses {
      ip_configuration_name = "primary-vpn"
      apipa_addresses       = var.vng_bgp_primary_ips
    }
    peering_addresses {
      ip_configuration_name = "secondary-vpn"
      apipa_addresses       = var.vng_bgp_secondary_ips
    }
  }
  tags = local.az_common_tags
  lifecycle {
    ignore_changes = [tags]
  }
}
```

### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v4.27.0_x5 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 133 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/network.expandVirtualNetworkGatewayVpnClientConfig(0xc0031b4140?, {{0xc0031b414f, 0x24}, {0xc0031b4183, 0x11}, {0xc0031b41c8, 0x11}})
github.com/hashicorp/terraform-provider-azurerm/internal/services/network/virtual_network_gateway_resource.go:1089 +0x1afb
github.com/hashicorp/terraform-provider-azurerm/internal/services/network.resourceVirtualNetworkGatewayUpdate(0xc0031a8e00, {0x7113bc0, 0xc0021c4488})
github.com/hashicorp/terraform-provider-azurerm/internal/services/network/virtual_network_gateway_resource.go:827 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x89bff08?, {0x89bff08?, 0xc00318f050?}, 0xd?, {0x7113bc0?, 0xc0021c4488?})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:860 +0x15f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0016b4e00, {0x89bff08, 0xc00318f050}, 0xc00182fa00, 0xc0031a8c80, {0x7113bc0, 0xc0021c4488})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:979 +0x84a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000b012a8, {0x89bff08?, 0xc00318ef90?}, 0xc0031903c0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/grpc_provider.go:1188 +0xd1a
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc000c22700, {0x89bff08?, 0xc00318ecc0?}, 0xc0031903c0)
github.com/hashicorp/terraform-plugin-mux@v0.17.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x18d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0009f59a0, {0x89bff08?, 0xc00318e1b0?}, 0xc001114070)
github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/tf5server/server.go:866 +0x3b9
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x7fbdc00, 0xc0009f59a0}, {0x89bff08, 0xc00318e1b0}, 0xc0031a8000, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:611 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001dea000, {0x89bff08, 0xc00318e150}, 0xc002b2e060, 0xc001ddafc0, 0xdb730b8, 0x0)
google.golang.org/grpc@v1.69.4/server.go:1392 +0xfbe
google.golang.org/grpc.(*Server).handleStream(0xc001dea000, {0x89c3ea8, 0xc001d561a0}, 0xc002b2e060)
google.golang.org/grpc@v1.69.4/server.go:1802 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
google.golang.org/grpc@v1.69.4/server.go:1030 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 10
google.golang.org/grpc@v1.69.4/server.go:1041 +0x11d

Error: The terraform-provider-azurerm_v4.27.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behaviour

Should have removed the VPN Client configuration as expected by the for_each

### Actual Behaviour

Plugin crashed.

### Steps to Reproduce

Proposed correct plan of removing the vpn client config, when applying crashes plugin.

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/network/virtual_network_gateway_resource.go
+++ b/internal/services/network/virtual_network_gateway_resource.go
@@ -1086,6 +1086,11 @@ func expandVirtualNetworkGatewayIPConfigurations(d *pluginsdk.ResourceData) *[]v
 
 func expandVirtualNetworkGatewayVpnClientConfig(d *pluginsdk.ResourceData, vnetGatewayId virtualnetworkgateways.VirtualNetworkGatewayId) *virtualnetworkgateways.VpnClientConfiguration {
 	configSets := d.Get("vpn_client_configuration").([]interface{})
+	if len(configSets) == 0 {
+		// return nil will delete the existing vpn client configuration
+		return nil
+	}
+
 	conf := configSets[0].(map[string]interface{})
 
 	confAddresses := conf["address_space"].([]interface{})

--- a/internal/services/network/virtual_network_gateway_resource_test.go
+++ b/internal/services/network/virtual_network_gateway_resource_test.go
@@ -168,6 +168,13 @@ func TestAccVirtualNetworkGateway_vpnClientConfig(t *testing.T) {
 				check.That(data.ResourceName).Key("vpn_client_configuration.0.vpn_client_protocols.#").HasValue("1"),
 			),
 		},
+		{
+			Config: r.deleteVPNClientConfig(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("vpn_client_configuration.#").HasValue("0"),
+			),
+		},
 	})
 }
 
@@ -905,6 +912,70 @@ uGLOhRJOFprPdoDIUBB+tmCl3oDcBy3vnUeOEioz8zAkprcb3GHwHAK+vHmmfgcn
 WsfMLH4JCLa/tRYL+Rw/N3ybCkDp00s0WUZ+AoDywSl0Q/ZEnNY0MsFiw6LyIdbq
 M/s/1JRtO3bDSzD9TazRVzn2oBqzSa8VgIo5C1nOnoAKJTlsClJKvIhnRlaLQqk=
 EOF
+    }
+  }
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, data.RandomInteger)
+}
+
+func (VirtualNetworkGatewayResource) deleteVPNClientConfig(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctestvn-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  address_space       = ["10.0.0.0/16"]
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "GatewaySubnet"
+  resource_group_name  = azurerm_resource_group.test.name
+  virtual_network_name = azurerm_virtual_network.test.name
+  address_prefixes     = ["10.0.1.0/24"]
+}
+
+resource "azurerm_public_ip" "test" {
+  name                = "acctestpip-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  allocation_method   = "Static"
+  sku                 = "Standard"
+}
+
+resource "azurerm_virtual_network_gateway" "test" {
+  depends_on          = [azurerm_public_ip.test]
+  name                = "acctestvng-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  type     = "Vpn"
+  vpn_type = "RouteBased"
+  sku      = "VpnGw1"
+
+  ip_configuration {
+    public_ip_address_id          = azurerm_public_ip.test.id
+    private_ip_address_allocation = "Dynamic"
+    subnet_id                     = azurerm_subnet.test.id
+  }
+
+  policy_group {
+    name       = "TestPolicyGroup"
+    is_default = true
+    priority   = 1
+
+    policy_member {
+      name  = "TestPolicyMember"
+      type  = "RadiusAzureGroupId"
+      value = "6ad1bd08"
     }
   }
 }

